### PR TITLE
Add docsy-example site for testing

### PR DESCRIPTION
**Pull request by kbhawkey**
Add `docsy-example` site for testing generated Markdown pages.
For some time, I have wanted a way (Hugo test site) to test the display of the generated documentation in this repository.
By using the example site and theme, the generated resources Markdown pages can be viewed without copying the content to a `website` branch. The site is for testing only and the content may not display identically to content in `website`.
There are build issues if the generated docs rely on a `website` configuration value or layouts/shortcodes.
For example, the generated _index.md page refers to the `glossary_definition` shortcode.
To get around this, I modified the Makefile to copy another _index.md file to the `reference/kubernetes-api` location. I suggest to remove this file from the resource generation. For more information, see issue #192.

### CODE REVIEW

**Comment by sftim** [comment]
Just now I've not got time to look at the implementation, but this sounds great.
/approve

**Review by irvifa — COMMENTED** [review]
/lgtm

File: docsy-example/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+Once you've cloned the site repo, from the repo root folder, run:
+
+```
+hugo server
+```
+
+### Running a container locally
+
+You can run docsy-example inside a [Docker](ihttps://docs.docker.com/)
```
Comment (sftim) [inline docsy-example/README.md:62]: nit:
```suggestion
You can run docsy-example inside a [Docker](https://docs.docker.com/)
```

**Comment by kbhawkey** [comment]
Whoops! I should have marked this PR as WIP. I planned to make several more changes.
Will create a follow-up PR.

File: docsy-example/README.md (hunk 1 above)
Comment (kbhawkey) [reply docsy-example/README.md:62]: I can check if I created this typo or if the upstream `README.md` has a typo (log an issue with the theme)?

Comment (sftim) [reply docsy-example/README.md:62]: Ah it might be that!

**Comment by kbhawkey** [comment]
I am making corrections and testing. Looks like the theme submodule is not set up in my last commit.

**Pull request merged**